code/datums/diseases/_disease.dm

```diff
@@ -46,7 +46,10 @@
 	infect(infectee, make_copy)
 	return TRUE
 
-//add the disease with no checks
+/**
+* add the disease with no checks
+* Don't use this proc. use ForceContractDisease on mob/living/carbon instead
+*/
 /datum/disease/proc/infect(var/mob/living/infectee, make_copy = TRUE)
 	var/datum/disease/D = make_copy ? Copy() : src	
```

Comment:
```suggestion
	PROTECTED_PROC(TRUE)
	var/datum/disease/D = make_copy ? Copy() : src	
```
Enforces not calling this proc outside of `/datum/disease` and its subtypes. If you think subtypes shouldn't be able to explicitly call it (or override it) then use `PRIVATE_PROC(TRUE)` instead. Honestly I'd try `PRIVATE_PROC(TRUE)` first and see if you get linter errors. If you don't, use it.

This change assumes that the proc doesn't violate that anywhere.

Reply:
alright, ill need to do it with try_infect as well